src/actions/queueaction/test_queueaction.ts

```diff
@@ -0,0 +1,39 @@
+import * as chai from "chai"
+import * as Hub from "../../hub"
+
+import {ExecuteProcessQueue} from "../../../lib/server/action_process_queue"
+import {QueueAction} from "./queueaction"
+
+const action = new QueueAction()
+const queue = new ExecuteProcessQueue()
+process.execArgv = ["./node_modules/ts-node/dist/bin.js"]
+process.argv = ["node", "./integrations/src/boot.ts"]
+process.env.CHILD_TEST = "true"
+
+function expectResponse(
+    request: Hub.ActionRequest,
+    success: boolean,
+) {
+    const execute = action.validateAndExecute(request, queue).then((response: Hub.ActionResponse) => {
+        chai.expect(response.success).to.equal(success)
+    })
+    return chai.expect(execute).to.be.fulfilled
+}
+
+describe(`${action.constructor.name} unit tests for ProcessQueue`, () => {
+    describe("action", () => {
+        it("returns success on successful response", () => {
+            const request = new Hub.ActionRequest()
+            request.type = Hub.ActionType.Query
+            request.attachment = {dataBuffer: Buffer.from(JSON.stringify({ success: true }))}
+            return expectResponse(request, true)
+        }).timeout(30000)
```

Comment:
is actually performing the fork consuming this amount of time?

Reply:
Yes it can

Reply:
Figuring out how to test this was a real pain but at least we have a template for it now